ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullFilterNode.java

```diff
@@ -78,9 +78,11 @@
   private final KsqlConfig ksqlConfig;
   private final LogicalSchema schema = getSource().getSchema();
 
+  // The rewritten predicate in DNF, e.g. (A AND B) OR (C AND D)
```

Comment:
I've made all of these fields final which should at least mitigate some risk that one will be modified in a random place.  They're written just once in the constructor so it's fairly easy to reason about.

I can make something that does the opposite of `LogicRewriter.extractDisjuncts` and joins together the disjuncts.  That way I can recreate rewrittenPredicate on demand, but honestly, that seems a little overkill and not worth the added complexity elsewhere.

Reply:
I do not feel strong about this (maybe I should use a `minor` label on comments next time), since they are now all `final` and written only once at constructor.